addons/pos_restaurant/models/pos_order.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    table_id = fields.Many2one('restaurant.table', string='Table', help='The table where this order was served')
+    customer_count = fields.Integer(string='Guests', help='The amount of customers that have been served by this order.')
+
+    @api.model
+    def _order_fields(self, ui_order):
+        fields = super(PosOrder, self)._order_fields(ui_order)
+        fields['table_id'] = ui_order.get('table_id', False)
+        fields['customer_count'] = ui_order.get('customer_count', False)
```

Comment:
Xavier told you to use `False` instead of `0` for `table_id` but for an Integer field, 0 is better :)